velox/type/TimestampConversion.h

```diff
@@ -107,6 +107,38 @@ Status daysSinceEpochFromWeekDate(
     int32_t dayOfWeek,
     int64_t& out);
 
+/// Computes the signed number of days since the Unix epoch (1970-01-01). To
+/// align with Spark's SimpleDateFormat behavior, this function offers two
+/// modes: lenient and non-lenient. If `lenient` is false, it returns an error
+/// status if the date is invalid. If `lenient` is true, it accepts a wider
+/// range of arguments.
+/// @param year Year. For non-lenient mode, it should be in the range [1,
+/// 292278994]. e.g: 1996, -2000. For lenient mode, values outside this range
+/// could result in overflow.
```

Comment:
> For lenient mode, values outside this range could result in overflow.

Will an exception be thrown, and do we have a unit test for this case? Thanks.

Reply:
@rui-mo In Java's SimpleDateFormat, if the year is greater than 292278994, the result will overflow (yielding a negative value). It is tricky to exactly replicate this behavior in Velox. Furthermore, I assume that the overflow behavior is not "correct" and is rarely useful. Therefore, I am inclined to implement a simpler behavior: either throw an exception or return a different result compared to Java's behavior. What do you think? Please let me know. Thanks.

Reply:
> either throw an exception or return a different result compared to Java's behavior

@NEUpanning Thanks for clarifying. Are we throwing exception for year overflow now? It makes sense to me.

Reply:
Thanks for the feedback. I added a unit test for this case.

Reply:
Let's make sure documentation is clear about this case.